controllers/TeacherExams.php

```diff
@@ -24,15 +27,46 @@ public function index()
     }
 
     public function uploadJson() {
+        $uploadOk = 1;
+        $fileName = basename($_FILES["fileToUpload"]["name"]);
+        $fileSize = basename($_FILES["fileToUpload"]["size"]);
+        $filePath = $_FILES["fileToUpload"]["tmp_name"];
+        $fileType = strtolower(pathinfo($fileName,PATHINFO_EXTENSION));
+
+        if ($fileSize > 5000) {
+            echo "Sorry, your file is too large.";
+            $uploadOk = 0;
+        }
+
+        if($fileType != "json") {
+            echo "Sorry, only JSON files are allowed.";
+            $uploadOk = 0;
+        }
+
+        if($uploadOk == 1) {
+            $myFile = fopen($filePath, "r") or die("Unable to open file!");
+            $jsonContent = fread($myFile,filesize($filePath));
+            fclose($myFile);
 
+            $exams = json_decode($jsonContent);
+            //TODO replace this with database insert
+            foreach ($exams->exams as $exam) {
+                echo $exam->name . " " . $exam->accessCode . "<br>";
+            }
+        }
     }
 
     public function downloadJson() {
-
+        $examsJson = json_encode(new ExamCollection($this->getUserExams()));
+        $fileName = "allExams" . date("Y-m-d-H-i-s") . ".json";
+        header('Content-disposition: attachment; filename=' . $fileName);
+        header('Content-type: application/json');
+        echo $examsJson;
     }
 
-    function getUserExams()
+    #[Pure] function getUserExams(): array
```

Comment:
this `#[Pure]` is crashes the flow, the following error is thrown: `syntax error, unexpected '{', expecting function (T_FUNCTION) or const (T_CONST)`

Reply:
without everything seems to be okay